ios/FlankerNativeComponents/GameManager.swift

```diff
@@ -184,271 +124,257 @@ class GameManager {
       case .feedback:
         endFeedbackTimestamp = time
       case .response:
-        respondTouchButton = time
+        break
       }
     }
   }
 
   func checkedAnswer(button: SelectedButton) {
+    guard !hasRespondedInCurrentTrial else { return }
+    hasRespondedInCurrentTrial = true
+    respondTouchButton = CACurrentMediaTime()
+    setEndTimeViewingImage(time: respondTouchButton!, isStart: false, type: .response)
```

Comment:
This call to `setEndTimeViewingImage` is not required, right? Right now it's only validating to `break` from the change in line 127

Reply:
I removed startTrialTimestamp in the setEndTimeViewingImage method for case .trial to prevent overwriting the startTrialTimestamp that is set elsewhere in your code. Previously, setting startTrialTimestamp here was causing issues with calculating the average response time because it was unintentionally overwriting the timestamp.

By setting startTrialTimestamp directly where it's needed (in the setText() method), we ensure that the correct timestamp is used for response time calculations. Removing it from setEndTimeViewingImage avoids conflicts and maintains accurate timing.

Reply:
You're correct. The call to setEndTimeViewingImage with type: .response is unnecessary because, in setEndTimeViewingImage. 
I will update it.